src/emulator/storage/persistence.ts

```diff
@@ -67,17 +71,34 @@ export class Persistence {
         if (err) {
           reject(err);
         } else {
+          this._diskPathMap = new Map();
           resolve();
         }
       });
     });
   }
 
   renameFile(oldName: string, newName: string): void {
-    renameSync(this.getDiskPath(oldName), this.getDiskPath(newName));
+    const oldNameId = this.getDiskFileName(oldName);
+    this._diskPathMap.set(newName, oldNameId);
+    this._diskPathMap.delete(oldName);
   }
 
   getDiskPath(fileName: string): string {
-    return path.join(this._dirPath, encodeURIComponent(fileName));
+    const shortenedDiskPath = this.getDiskFileName(fileName);
+    return path.join(this._dirPath, encodeURIComponent(shortenedDiskPath));
+  }
+
+  getDiskFileName(fileName: string): string {
+    return this._diskPathMap.get(fileName)!;
+  }
+
+  copyFromExternalPath(sourcePath: string, newName: string): void {
```

Comment:
test?

Reply:
Will add a unit test for the new methods